components/scream/src/share/io/scorpio_input.cpp

```diff
@@ -52,14 +66,12 @@ void AtmosphereInput::pull_input()
   init();
   for (auto const& name : m_fields)
   {
-    auto fmap = m_field_repo->get_field(name, m_grid_name);
+    auto field = m_field_repo->get_field(name, m_grid_name);
     // Get all the info for this field.
-    auto l_view_dev = fmap.get_view();
-    view_type l_view = m_view_local.at(name);
-    auto l_dims = fmap.get_header().get_identifier().get_layout().dims();
-    Int padding = fmap.get_header().get_alloc_properties().get_padding();
-    grid_read_data_array(m_filename,name,l_dims,m_dofs.at(name),padding,l_view.data());
-    Kokkos::deep_copy(l_view_dev,l_view);
+    auto l_dims = field.get_header().get_identifier().get_layout().dims();
+    Int padding = field.get_header().get_alloc_properties().get_padding();
+    grid_read_data_array(m_filename,name,l_dims,m_dofs.at(name),padding,field.get_view<Host>().data());
```

Comment:
Sorry, one more comment: if `field.get_view<Host>` returns a strided view, then passing the pointer is dangerous.

We can think about how to deal with strided views (like a single tracer) later. Meanwhile, we could put a check to make sure the field is not strided. The check would look like
```
EKAT_REQUIRE_MSG (field.get_header().get_alloc_properties().contiguous(), "Error! blah blah\n");
```

Reply:
Oh yes, great point.  We will need to figure out how to handle this, and maybe get rid of `get_view` calls everywhere.  Probably a good git issue, so we don't lose sight of the goal.